File: src/lib/library/v1/heightshift.ts
import { GpuDesignerNode } from "../../designer/gpudesignernode";

export class HeightShiftNode extends GpuDesignerNode {
	public init() {
		this.title = "Height Shift";

		this.addInput("image");

		this.addFloatProperty("shift", "Shift", 0.0, -1.0, 1.0, 0.01);

		const source = `
        vec4 process(vec2 uv)
        {
            vec4 a =  texture(image, uv);

            return a + vec4(vec3(prop_shift), 0.0);
        }
        `;

		this.buildShader(source);
	}
}

File: src/texturelab/libraries/v1/heightshift.cpp
#include "../../models.h"
#include "../../props.h"
#include "../libv1.h"

void HeightShiftNode::init()
{
    this->title = "Height Shift";

    this->addInput("image");

    this->addFloatProp("shift", "Shift", 0.0, -1.0, 1.0, 0.01);

    auto source = R""""(
        vec4 process(vec2 uv)
        {
            vec4 a =  texture(image, uv);

            return a + vec4(vec3(prop_shift), 0.0);
        }
        )"""";

    this->setShaderSource(source);
}